prov/efa/src/efa_hmem.c

```diff
@@ -298,6 +302,54 @@ static int efa_hmem_info_init_synapseai(struct efa_hmem_info *synapseai_info, st
 	return 0;
 }
 
+/**
+ * @brief   Validate an FI_OPT_FI_HMEM_P2P (FI_OPT_ENDPOINT) option for a
+ *          specified HMEM interface
+ *
+ * @param   domain  The efa_domain struct which contains an efa_hmem_info array
+ * @param   iface   The fi_hmem_iface enum of the HMEM interface to validate
+ * @param   p2p_opt The P2P option to validate
+ *
+ * @return  FI_SUCCESS if the P2P option is valid for the given interface
+ *         -FI_OPNOTSUPP if the P2P option is invalid
+ *         -FI_ENODATA if the given HMEM interface was not initialized
+ *         -FI_EINVAL if p2p_opt is not a valid FI_OPT_FI_HMEM_P2P option
+ */
+int efa_hmem_validate_p2p_opt(struct efa_domain *efa_domain, enum fi_hmem_iface iface, int p2p_opt)
```

Comment:
should be `set_p2p_opt` because it does not only validate but also set `disabled_by_user`.

Reply:
`p2p_opt` here refers to the `fi_setopt()`/`FI_OPT_FI_HMEM_P2P` options, which is what's being validated in this function. This function is appropriately named. Including `set` in the function name implies setting of the user opt. The caller doesn't need to be made aware the function modifies internal `efa_hmem_info` state as a result of `FI_HMEM_P2P_DISABLED`.

Reply:
>The caller doesn't need to be made aware the function modifies internal efa_hmem_info state as a result of FI_HMEM_P2P_DISABLED.

I am not sure. the caller of `fi_setopt` does not need to know `efa_hmem_info` has been updated. But this function is called by an internal function.

I understand the behavior might not be easy to be reflected in function name. Can you added the behavior (updating disabled_by_user) to the doxgen header of this function?